packages/plugin-compiler/src/main/kotlin/io/realm/compiler/AccessorModifierIrGeneration.kt

```diff
@@ -430,36 +430,53 @@ class AccessorModifierIrGeneration(private val pluginContext: IrPluginContext) {
         return propertyClassId == realmListClassId
     }
 
-    private fun getListGenericType(declaration: IrProperty): List<String> {
+    private fun getListGenericCoreType(declaration: IrProperty): CoreType {
         // Check first if the generic is a subclass of RealmObject
-        val superTypes = declaration.symbol.descriptor.type.arguments[0].type.constructor.supertypes
-        for (superType: KotlinType in superTypes) {
+        val descriptorType = declaration.symbol.descriptor.type
+        val listGenericType = descriptorType.arguments[0].type
+        for (superType in listGenericType.constructor.supertypes) {
             if (superType.toString() == "RealmObject") {
-                return listOf("RealmObject")
+                // Nullable objects are not supported
```

Comment:
Do we have the same constraint on Realm-Java? I know it's a Core limitation but I was under the impression it was still possible ... not sure when this constraint was added ... cc @cmelchior

Reply:
Yes, RealmLists that contain RealmObjects are not allowed to contain `null` values. I actually do not remember if RealmLists of embedded objects are allowed.

Reply:
@finnschiermer is there an issue tracking the above? to keep consistency with other supported types, I think `RealmList` should allow nullable `RealmObject` ?

Reply:
There is no such issue, the "anomaly" in how lists handle (or rather does not handle) null references is by design. I have been led to believe over time that some users rely on (or even want) this behavior. I don't. Feel free to open such an issue to start a discussion.

Reply:
@finnschiermer Java lists (and collection API) supports adding `null` value(s). Since our goal is to be as close as possible to the platform's language we should support this use case. This also needs to be unified across other data structures like Set and Map.
Another argument is persisting an unmanaged RealmList (which contains already nullable elements coming from JSON for instance) into a managed one, since List is ordered the persisted RealmLIst should reflect the original order/elements

Reply:
I've created https://github.com/realm/realm-kotlin/issues/292 to follow up on this.